src/Xamarin.Android.Build.Tasks/Utilities/ResourceIdentifier.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.CodeDom;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	public class ResourceIdentifier {
+
+		//https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+		private const string FormattingCharacter = @"\p{Cf}";
+		private const string ConnectingCharacter = @"\p{Pc}";
+		private const string DecimalDigitCharacter = @"\p{Nd}";
+		private const string CombiningCharacter = @"\p{Mn},\p{Mc}";
+		private const string LetterCharacter = @"\p{Lu},\p{Ll},\p{Lt},\p{Lm},\p{Lo},\p{Nl}";
+
+		private const string IdentifierPartCharacter = LetterCharacter + "," +
+			DecimalDigitCharacter + "," +
+			ConnectingCharacter + "," +
+			CombiningCharacter + "," +
+			FormattingCharacter;
+
+		private const string IdentifierStartCharacter = "(" + LetterCharacter + ",_)";
+
+		private const string Identifier = IdentifierStartCharacter + "((" + IdentifierPartCharacter + ")+)*";
+
+		//https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#keywords
```

Comment:
If we're going to "blacklist" identifiers, then we should also blacklist *Java* identifiers: https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.9

Reply:
Not sure that will work. Its only the C# identifiers we need to prefix with a `@`.

Reply:
Do we just need to make sure there is an `@` symbol for these?

For example, I can do:
```
public const int @class = 0;
```

Reply:
@jonathanpeppers that is handled by https://github.com/xamarin/xamarin-android/pull/3025/files#diff-fe72c1d40099e8bd2cfa00d8eeccad39R63